README.md

```diff
@@ -1,13 +1,49 @@
+[![Build Status](https://travis-ci.org/IBM/serverless-language-translation.svg?branch=master)](https://travis-ci.org/IBM/serverless-language-translation)
+
 # serverless-language-translation
 
-This repository contains a series of serverless functions that can serve as the backend for a multilingual chat room, as described in the blog post [here](https://medium.com/kkbankol-events/the-motivation-behind-this-particular-project-comes-from-playing-one-of-my-favorite-android-games-76c92b27c8e8)
+For this code pattern, this repository contains a series of serverless functions that can serve as the backend for a multilingual chat room.  
+
+When the reader has completed this Code Pattern, they will understand how to:
+
+* deploy Cloud Function actions and triggers
+* interact with the Watson IoT platform
+* set up a CloudFoundry application
 
 ![Architecture](/assets/architecture.png)
-TODO, update architecture
 
-Prerequisites:
+## Flow
+
+1. Message received from a client, which can be a web browser, CLI, Openwhisk action, SMS text, etc.
+2. If message payload contains an audio file, it is transcribed to text.
+3. Transcribed text is translated to other supported languages.
+4. If message sent via SMS, sender phone number is added to an etcd key/value store. etcd is used here to maintain a list of subscriber’s phone numbers, as well as their respective languages. An adjustable TTL value is used here to remove numbers from the store if the subscriber does not participate in the conversation for 300 seconds.
+5. Translated messages/audio streams are published to various channels on the MQTT broker, which then distributes the messages amongst subscribing clients.
+
+## Included components
+
+* [OpenWhisk](https://console.ng.bluemix.net/openwhisk): Execute code on demand in a highly scalable, serverless environment.
+* [Watson Text to Speech](https://www.ibm.com/watson/developercloud/text-to-speech.html): Converts written text into natural sounding audio in a variety of languages and voices.
+
+## Featured technologies
+
+* [Messaging](https://developer.ibm.com/messaging/message-hub/): Messaging is a key technology for modern applications using loosely decoupled architecture patterns such as microservices.
+* [Node.js](https://nodejs.org/): An open-source JavaScript run-time environment for executing server-side JavaScript code.
+* [Serverless](https://www.ibm.com/cloud-computing/bluemix/openwhisk): An event-action platform that allows you to execute code in response to an event.
+
+# Watch the Video
+![](https://imgur.com/Q4DGOPM.mp4)
```

Comment:
this isn't going to be the final video right?

Reply:
that sounds more like a statement than a question... :)